src/utils/functions.js

```diff
@@ -103,46 +106,80 @@ const renderProperties = (param) => {
 
 // eslint-disable-next-line enact/prop-types
 const Parameters = ({func, params, hasReturns}) => {
+	const [paramType, setParamType] = useState({});
+	const [methodReturnValue, setMethodReturnValue] = useState([]);
+	const [responseParamCountString, setResponseParamCountString] = useState([]);
+	const [responseRenderProperties, setResponseRenderProperties] = useState({});
+
+	useEffect(() => {
+		const renderParamCountString = async () => {
+			const data = await paramCountString(params);
+			setResponseParamCountString(data);
+		};
+		renderParamCountString()
+			.catch(console.error); // eslint-disable-line no-console
+
+		// map over all parameters, extract their type and render properties inside an object with the name of the property/type as the key
+		const renderParamTypeAndPropertiesEffect = Promise.all(params.map(async (param) => {
+			const paramTypeData = await renderTypeStrings(param);
+			setParamType(obj => Object.assign(obj, {[param.name]: paramTypeData}));
+
+			const renderPropertiesData = await renderProperties(param);
+			setResponseRenderProperties(obj => Object.assign(obj, {[param.name]: renderPropertiesData}));
+		}));
+		renderParamTypeAndPropertiesEffect
+			.catch(console.error); // eslint-disable-line no-console
+
+		// get the return value of any methods
+		const renderMethodReturnValue = async () => {
+			const data = await renderTypeStrings(func.returns);
+			setMethodReturnValue(array => [...array, data]);
+		};
+		renderMethodReturnValue()
+			.catch(console.error); // eslint-disable-line no-console
+
+	}, [func.returns, params]);
+
```

Comment:
Note: `Parameters` function component calls `renderTypeStrings`, `paramCountString` and `renderProperties` which return a promise. We cannot declare a function component
as async/await, so I stored all the values the above functions return in a state and configured them in `useEffect`.